### PR TITLE
Un-tag and Prune TR Images properly

### PR DESCRIPTION
**Pull request by tylertitsworth**
## Description
Do what we do in the Build CI

## Related Issue
n/a

## Changes Made

- [ ] The code follows the project's [coding standards](../CONTRIBUTING.md#code-style).
- [ ] No Intel Internal IP is present within the changes.
- [ ] The documentation has been updated to reflect any changes in functionality.

## Validation
See summary of: https://github.com/intel/ai-containers/actions/runs/9456527259
Container there is from a cancelled job.

- [ ] I have tested any changes in container groups locally with [`test_runner.py`](../test-runner/README.md) with all existing tests passing, and I have added new tests where applicable.

### CODE REVIEW

File: test-runner/test_runner.py [hunk 1]
```diff
@@ -205,7 +205,8 @@ def get_test_list(args: dict, tests_yaml: List[dict]):
         remaining_containers = docker.container.list()
         for container in remaining_containers:
             docker.stop(container, time=None)
-        docker.image.remove(test_images, force=True)
+        docker.image.remove(test_images, force=True, prune=False)
```
Comment (sharvil10) [inline test-runner/test_runner.py:208]: Any particular reason why we need a separate prune and can't use the prune param in this function?

Comment (tylertitsworth) [reply test-runner/test_runner.py:208]: It leaves behind some images, and causes our GPU runners to run out of space after multiple preset `test-containers` jobs.

Comment (tylertitsworth) [reply test-runner/test_runner.py:208]: This is a regression in `python_on_whales`. I validated that issue locally and say that by running system prune it doesn't leave the same images behind.

**Review by sharvil10 — APPROVED** [review]
LGTM

**Pull request merged**